docs/guide/ibc.md

```diff
@@ -0,0 +1,543 @@
+
+Append the type instance as `PostID` on receiving the packet:
+
+- The context `ctx` is an [immutable data structure](https://docs.cosmos.network/master/core/context.html#go-context-package) that has header data from the transaction. See [how the context is initiated](https://github.com/cosmos/cosmos-sdk/blob/master/types/context.go#L71)
+- The identifier format that you defined earlier
+- The `title` is the Title of the blog post
+- The `content` is the Content of the blog post
+
+In the `ibcPost.go` file, make sure to import `"strconv"` below `"errors"`, and then modify `OnRecvIbcPostPacket` with the following code:
+
+```go
+// x/blog/keeper/ibc_post.go
+import (
+  //...
+  "strconv"
+)
+
+func (k Keeper) OnRecvIbcPostPacket(ctx sdk.Context, packet channeltypes.Packet, data types.IbcPostPacketData) (packetAck types.IbcPostPacketAck, err error) {
+    // validate packet data upon receiving
+    if err := data.ValidateBasic(); err != nil {
+        return packetAck, err
+    }
+    id := k.AppendPost(
+        ctx,
+        types.Post{
+            Creator: packet.SourcePort+"-"+packet.SourceChannel+"-"+data.Creator,
+            Title: data.Title,
+            Content: data.Content,
+        },
+    )
+    packetAck.PostID = strconv.FormatUint(id, 10)
+    return packetAck, nil
+}
+```
+
+### Receive the post acknowledgement
+
+On the sending blockchain, store a `sentPost` so you know that the post has been received on the target chain.
+
+Store the title and the target to identify the post.
+
+When a packet is scaffolded, the default type for the received acknowledgment data is a type that identifies if the packet treatment has failed. The `Acknowledgement_Error` type is set if `OnRecvIbcPostPacket` returns an error from the packet.
+
+```go
+// x/blog/keeper/ibc_post.go
+func (k Keeper) OnAcknowledgementIbcPostPacket(ctx sdk.Context, packet channeltypes.Packet, data types.IbcPostPacketData, ack channeltypes.Acknowledgement) error {
+	switch dispatchedAck := ack.Response.(type) {
+	case *channeltypes.Acknowledgement_Error:
+		// We will not treat acknowledgment error in this tutorial
+		return nil
+	case *channeltypes.Acknowledgement_Result:
+		// Decode the packet acknowledgment
+		var packetAck types.IbcPostPacketAck
+
+		if err := types.ModuleCdc.UnmarshalJSON(dispatchedAck.Result, &packetAck); err != nil {
+			// The counter-party module doesn't implement the correct acknowledgment format
+			return errors.New("cannot unmarshal acknowledgment")
+		}
+		k.AppendSentPost(
+			ctx,
+			types.SentPost{
+				Creator: data.Creator,
+				PostID:  packetAck.PostID,
+				Title:   data.Title,
+				Chain:   packet.DestinationPort + "-" + packet.DestinationChannel,
+			},
+		)
+		return nil
+	default:
+		// The counter-party module doesn't implement the correct acknowledgment format
+		return errors.New("invalid acknowledgment format")
```

Comment:
```suggestion
		return errors.New("The counter-party module does not implement the correct acknowledgment format")
```

Reply:
As far as I remember, errors should be in lower case.